### PR TITLE
create wntr object directly from WaterModels output

### PR DESCRIPTION
**Pull request by enzoliuyang94**
Implemented the direct mapping from WaterModels to a WNTR network object that is then simulated. Two comments (the second one way more important):

1. I am not sure what is 'regulator' and currently do not consider it in 'simulation.jl' and 'validation.jl'
2. This is particularly relevant if we call WMA during the optimization process - even if we enforce controls to WNTR, when it actually simulates, WNTR could CHANGE some controls to make a set of otherwise infeasible controls feasible. We can catch this and call such controls infeasible, but this seems to be happening even in the optimized pumping schedule (an example is sent via E-mail to you)

### CODE REVIEW

**Comment by tasseff** [comment]
Regarding your first comment, a regulator is the same as a "pressure reducing valve" in EPANET. Regarding the second point, I just responded to your email. I will try to test and thoroughly review this code tomorrow, but from a cursory look, it looks very intuitive!

File: src/analysis/simulation.jl [hunk 1]
```diff
@@ -7,172 +7,202 @@ wm_solution.
...
+        else
+            wn.options.energy.global_efficiency = pump_info["efficiency"]
+        end
     end
 
+
+
+
+
     # remove old controls
```
Comment (jjstickel) [inline src/analysis/simulation.jl:206]: If we are creating the WNTR network from scratch, I don't think there are any old controls to remove?

File: src/analysis/validation.jl [hunk 2]
```diff
@@ -76,88 +67,27 @@ function get_tank_dataframe(wm_data,wm_solution,wntr_data,wntr_simulation, tank_
...
-
-    link_df = DataFrames.DataFrame(time = 1:time_step:time_step*num_time_step, flow_wntr = flow_wntr, flow_watermodels = flow_watermodels, 
-        head_loss_wntr = head_loss_wntr, head_loss_watermodels = head_loss_watermodels)
-
-    return link_df
-end
 
 # compare pipes
 function get_pipe_dataframe(wm_data,wm_solution,wntr_data,wntr_simulation,pipe_id)
     num_time_step = length(wm_solution["solution"]["nw"])            # number of time steps
```
Comment (jjstickel) [inline src/analysis/validation.jl:73]: There is a lot of duplicated code blocks for each of these functions `get_pipe_dataframe`, `get_short_pipe_dataframe`, and `get_valve_dataframe`. You might consider making a function (or two) that could execute these blocks with a keyword to designate the link type. Just a suggestion.

**Review by jjstickel — APPROVED** [review]
Consider removing the two unnecessary blocks of code I identified. Otherwise, this looks good to me!

File: src/analysis/simulation.jl (hunk 1 above)
Comment (enzoliuyang94) [reply src/analysis/simulation.jl:206]: @jjstickel Thank you for these comments, I will address them and make another commit.

**Pull request merged**